mtp_cashbook/apps/mtp_auth/forms.py

```diff
@@ -7,31 +8,63 @@
 from mtp_common.user_admin.forms import SignUpForm
 from oauthlib.oauth2 import OAuth2Error
 from requests import RequestException
+from zendesk_tickets.forms import BaseTicketForm
 
 logger = logging.getLogger('mtp')
 
 ACCOUNT_REQUEST_ROLE = 'prison-clerk'
 
 
-class CashbookSignUpForm(SignUpForm):
+class CashbookSignUpForm(SignUpForm, BaseTicketForm):
     prison = forms.ChoiceField(label=_('Prison'), help_text=_('Enter the prison you’re based in'))
 
+    # Non form class variables
+    cashbook_account_request_zendesk_subject = 'Requesting a new staff account - Digital Cashbook'
+    zendesk_tags = ('mtp', 'cashbook', 'account_request')
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         role_field = self.fields['role']
         role_field.initial = ACCOUNT_REQUEST_ROLE
         role_field.choices = ((ACCOUNT_REQUEST_ROLE, 'Business hub'),)
         prison_field = self.fields['prison']
-        prison_field.choices = [['', _('Select a prison')]] + self.prison_choices
+        prison_field.choices = [['', _('Select a prison')]] + list(self.prison_choices.items())
 
     @property
     def prison_choices(self):
         choices = cache.get('sign-up-prisons')
         if not choices:
             try:
                 choices = retrieve_all_pages_for_path(self.api_session, '/prisons/', exclude_empty_prisons=True)
-                choices = [[prison['nomis_id'], prison['name']] for prison in choices]
+                choices = OrderedDict([[prison['nomis_id'], prison['name']] for prison in choices])
                 cache.set('sign-up-prisons', choices, timeout=60 * 60 * 6)
             except (RequestException, OAuth2Error, ValueError):
                 logger.exception('Could not look up prison list')
         return choices
+
+    def user_already_requested_account(self):
+        # TODO Also, is it worth adding a unique_together definition for username/role fields?
```

Comment:
do you mean on the model? i _think_ maybe it's best to not cos users shouldn't be allowed access to multiple apps (which is a slightly weird limitation)

Reply:
Happy to leave it for now, but my take is that adding that constraint (and an appropriate error message) would mean that a user can't request access to multiple apps at the same time.